src/Event/Subscriber/MutationTestingResultsCollectorSubscriber.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Event\Subscriber;
+
+use Infection\Event\MutantProcessWasFinished;
+use Infection\Metrics\Collector;
+
+/**
+ * @internal
+ */
+final class MutationTestingResultsCollectorSubscriber implements EventSubscriber
+{
+    /** @var Collector[] */
```

Comment:
```suggestion
    /** @var array<int, Collector> */
```

?

Reply:
We're not using the key anywhere, so more appropriate definition would be `array<Collector>`. It is also saves us from the trouble declaring the array-key type for PHPStan up the stream. And, well, `array<Collector>` is same as `Collector[]`.

Reply:
Again, key-type is insignificant here.

Reply:
but what kind of integer keys are there? If they are natural numbers, we can use `list<Collector>`, if they are integers with "wholes", it would be better to use `array<int, Collector>`

IMO, `int` is better than implicit `array-key`

Reply:
What I mean is that the API of `MutationTestingResultsCollectorSubscriber` does not care about keys.

`iterable<Collector>` could work.